src/nix-daemon/nix-daemon.cc

```diff
@@ -277,7 +278,9 @@ static int _main(int argc, char * * argv)
         initPlugins();
 
         if (stdio) {
-            if (getStoreType() == tDaemon) {
+            if (openUncachedStore().dynamic_pointer_cast<UDSRemoteStore>()) {
```

Comment:
Doesn't this cause the store to be opened twice in the non-UDSRemoteStore case? Maybe it's better to do `store = openUncachedStore(); if (store.dynamic_pointer_cast ...) ... `.

Reply:
@edolstra Ah I remember now: I wanted to do that, but I wasn't sure how to destroy the opened store in the `UDSRemoteStore` before reopening it, since `nix::ref` is always non-null.

Reply:
The connections are opened on demand, so this was never a problem, but the new way also uses the store that was opened/downcast so there is no duplicate work.